src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/persistence/SQLitePersistor.java

```diff
@@ -187,6 +192,35 @@ synchronized String readTables() {
     return tablesObject.toString();
   }
 
+  public synchronized List<Result<Record>> getRecordsForAllTables() {
```

Comment:
We removed this in the latest revision. Because the REST API queries the DB to return the result. Therefore, we don't need this. This will simplify, what a validator has to override.

Reply:
To answer the question, this is required because a the SQLite will synchronize access to DB by default (unless you ask it not to bother about it). In our cases, we don't override so writes to DB are synchronized. But we do something that can lead to errors. We, rotate the DB file with  certain periodicity. SO it might so happen that when we wanted to read, the DB context was invalid and we will error out.